controllers/infra/drivers/aliyun/create_keypair.go

```diff
@@ -0,0 +1,55 @@
+package aliyun
+
+import (
+	"fmt"
+
+	"github.com/labring/sealos/pkg/utils/logger"
+
+	"github.com/labring/sealos/controllers/infra/common"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	v1 "github.com/labring/sealos/controllers/infra/api/v1"
+)
+
+type ECSCreateKeyPairAPI interface {
+	CreateKeyPair(request *ecs.CreateKeyPairRequest) (response *ecs.CreateKeyPairResponse, err error)
+}
+
+func CreateKeyPair(api ECSCreateKeyPairAPI, request *ecs.CreateKeyPairRequest) (*ecs.CreateKeyPairResponse, error) {
+	return api.CreateKeyPair(request)
+}
+
+func (d Driver) createKeyPair(infra *v1.Infra) error {
+	client := d.Client
+	keyPairName := getKeyPairName(infra)
+	keyPairTag := getKeyPairTag(infra)
+	createKeyPairRequest := &ecs.CreateKeyPairRequest{
+		RpcRequest:  ecs.CreateCreateKeyPairRequest().RpcRequest,
+		KeyPairName: keyPairName,
+		Tag:         keyPairTag,
+	}
+
+	result, err := CreateKeyPair(client, createKeyPairRequest)
+	if err != nil {
+		return fmt.Errorf("failed to create keypair %s: %v", infra.UID, err)
+	}
+	infra.Spec.SSH.PkName = result.KeyPairName
+	infra.Spec.SSH.PkData = result.PrivateKeyBody
+	logger.Info("Created keypair %s success", keyPairName)
+	return nil
+}
+
+func getKeyPairName(infra *v1.Infra) string {
+	return fmt.Sprintf("%s-%s", common.AliyunKeyPairPrefix, infra.UID)
```

Comment:
`KeypairName` is related to infra.uid, so all infras keypairname is same across all clusters under one user?
Or one cluster one keypair?

Reply:
> 

every infra's uid is unique, and one infra relates to one cluster, so one cluster use one keypair.

Reply:
> `KeypairName` is related to infra.uid, so all infras keypairname is same across all clusters under one user? Or one cluster one keypair?

every infra's uid is unique, and one infra relates to one cluster, so one cluster use one keypair.